src/utils.js

```diff
@@ -12,3 +13,21 @@ export function getDataHubParams(partialUrl) {
     search: parsedUrl.query,
   };
 }
+
+export function useInterval(callback, delay) {
+  const savedCallback = useRef();
+
+  useEffect(() => {
```

Comment:
I will try to understand why is this _useEffect_ necessary.

Reply:
Because you need some way of making the callback update, else you get the state frozen like the first time you've entered `useInverval` hook. The mechanism behind this, I'm not so sure.

Reply:
No, creo que la respuesta no va por ahí, porque en ese caso, esto también funcionaría:

```diff
- useEffect(() => {
   savedCallback.current = callback;
- }, [callback]);
```

El código dentro del `useEffect` se ejecuta después del código principal, pero aún no entiendo cual es la ventaja.

Reply:
Maybe they can help us: https://github.com/gaearon/overreacted.io/pull/346#issuecomment-518311802